sqlglot/generator.py

```diff
@@ -12,7 +12,17 @@
 logger = logging.getLogger("sqlglot")
 
 
-class Generator:
+class _Generator(type):
+    def __new__(cls, clsname, bases, attrs):
+        klass = super().__new__(cls, clsname, bases, attrs)
+
+        klass._WITH_SEPARATED_COMMENTS = tuple(klass.WITH_SEPARATED_COMMENTS)
```

Comment:
what's the point of this?

Reply:
Mostly trying to respect type constraints, as overriding a tuple attribute to create a tuple of different type (or cardinality, in this case) in a subclass is not a valid form and mypy yells. Happy to revert the change and throw some `# type: ignore`s in there if you think it's unnecessary overhead / complexity.

Reply:
why do you need tuples? just keep it a set?

Reply:
We need tuples because these expression types are used in instance checks

Reply:
just make the class WITH_SEPARATE_COMMENTS: t.Tuple[t.Type[exp.Expression], ...]

Reply:
Yep, tried that unfortunately it also didn't solve the problem:

```
sqlglot/dialects/duckdb.py:238: error: Incompatible types in assignment (expression has type "Tuple[Type[Expression], ...]", base class "Generator" defined the type as "Tuple[Type[Column], Type[Literal], Type[Neg], Type[Paren]]")  [assignment]